ha/core/cluster/cluster_manager.py

```diff
@@ -180,9 +172,81 @@ def add_node(self, node):
         _output, _err, _rc = self._execute.run_cmd(const.PCS_STATUS)
         Log.info(f"Cluster status output after add node: {_output}, {_err}, {_rc}")
 
+    def get_nodes_status(self):
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_STATUS_NODES, check_error=False)
+
+        self.active_nodes = self.standby_nodes = self.offline_nodes = "false"
+        for status in _output.split("\n"):
+            nodes = status.split()
+            if len(nodes) > 0:
+                if nodes[0] == "Online:" and len(nodes) > 1:
+                    self.active_nodes = "true"
+
+                if nodes[0] == "Standby" and nodes[1] == "with" and nodes[2] == "resource(s)" and nodes[3] == "running:" and len(nodes) > 4:
+                    self.active_nodes = "true"
+ 
+                if((nodes[0] == "Maintenance:") and (len(nodes) > 1)):
+                    self.active_nodes = "true"
+
+                if((nodes[0] == "Standby:") and (len(nodes) > 1)):
+                    self.standby_nodes = "true"
+
+                if((nodes[0] == "Offline:") and (len(nodes) > 1)):
+                    self.offline_nodes = "true"
+
+
     def start(self):
-        # TODO Add wrapper to hctl pcswrap
-        raise HAUnimplemented("This feature is not supported...")
+
+        Log.debug("Executing cortxha cluster start")
+
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_CLUSTER_STATUS, check_error=False)
+        if _rc != 0:
+            if(_err.find("No such file or directory: 'pcs'") != -1):
+                Log.error("Cluster failed to start; pcs not installed ")
+                raise Exception(f"Cluster failed to start; pcs not installed")
+            # if cluster is not running; start cluster
+            elif(_err.find("cluster is not currently running on this node") != -1):
+                output, _err, _rc = self._execute.run_cmd(const.PCS_CLUSTER_START, check_error=False)
+                Log.info(f"cluster started ; waiting for nodes to come online ")
+                # It takes nodes 30 seconds to come to their original state after cluster is started
+                # observation on a 2 node cluster
+                time.sleep(30)
```

Comment:
Can this sleep time be configurable?

Reply:
@Indrajit, we have made this value 100 seconds and checks happening at every 5 second interval